api/clients/v2/relay_registry_caller.go

```diff
@@ -0,0 +1,65 @@
+package clients
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/common/geth"
+	registry_binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDARelayRegistry"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Provides accessor methods for read interfacing with
+// relay registry contract (core/EigenDARelayRegistry.sol)
+type RelayRegistryCaller interface {
+	// Reads the current sockets or key->url mapping
+	// from the on-chain registry
+	GetSockets() (map[corev2.RelayKey]string, error)
```

Comment:
I don't think this should be called "GetSockets". I know that's the terminology used in the `relay_client`, but these are URLs, not sockets. Possible I'm wrong, but I've never heard a URL be called a "socket"

Reply:
They use Socket to mean URL:port combination. Not sure whether relays register a full socket address or just a URL where default 443 port or something is used though.

Reply:
I would call that an "endpoint". Socket really just seems like the wrong term

Reply:
https://www.ibm.com/docs/en/zvse/6.2?topic=considerations-socket-addresses

Reply:
`Socket` != `SocketAddress`. *Maybe* you could argue this map contains socket addresses, but certainly not sockets

`SocketAddress` does indeed contain an address and a port, but it seems to me like the term is used internally on a machine, to reference the literal sockets that have been created locally.

Reply:
`socket` generally has a technical meaning in networking, and an `IP_ADDRESS:PORT` string isn't a socket (although it does tell you how to open a socket). I think we should be careful about how we use the term `socket`.

Reply:
I also didn't like the term originally, and did a bunch of googling lol. Seems like its an accepted term? https://en.wikipedia.org/wiki/Network_socket#Socket_addresses

I personally find endpoint to not be precise enough, but agree that `socket` by itself is super confusing and nor is it anymore precise.

Reply:
It's *an* accepted term, but I don't think in this context. From the linked article:

```
This combination is often known as a socket address. It is the network-facing access handle to the network socket.
```

It's used in a context specifically related to interactions with network sockets.